File: Models/Stics_soil_temperature/src/cs/Stics_soil_temperature/soil_tempComponent.cs
public class soil_tempComponent
{
    
        public soil_tempComponent() { }
    

    //Declaration of the associated strategies
    Temp_amp _Temp_amp = new Temp_amp();
    Temp_profile _Temp_profile = new Temp_profile();
    Layers_temp _Layers_temp = new Layers_temp();
    Canopy_temp_avg _Canopy_temp_avg = new Canopy_temp_avg();
    Update _Update = new Update();

    public double air_temp_day1
    {
        get
        {
             return _temp_profile.air_temp_day1; 
        }
        set
        {
            _temp_profile.air_temp_day1 = value;
        }
    }
    public int[] layer_thick
    {
        get
        {
             return _temp_profile.layer_thick; 
        }
        set
        {
            _temp_profile.layer_thick = value;
            _layers_temp.layer_thick = value;
        }
    }

    public void  CalculateModel(soil_tempState s, soil_tempState s1, soil_tempRate r, soil_tempAuxiliary a, soil_tempExogenous ex)
    {
        _Temp_amp.CalculateModel(s,s1, r, a, ex);
        _Canopy_temp_avg.CalculateModel(s,s1, r, a, ex);
        _Temp_profile.CalculateModel(s,s1, r, a, ex);
        _Layers_temp.CalculateModel(s,s1, r, a, ex);
        _Update.CalculateModel(s,s1, r, a, ex);
    }
    
    public soil_tempComponent(soil_tempComponent toCopy): this() // copy constructor 
    {

        air_temp_day1 = toCopy.air_temp_day1;
        
            for (int i = 0; i < 100; i++)
            { layer_thick[i] = toCopy.layer_thick[i]; }
    
    }
}

File: Models/Stics_soil_temperature/src/cpp/Stics_soil_temperature/soil_tempComponent.cpp
#include "soil_tempComponent.h"
using namespace Stics_soil_temperature;
soil_tempComponent::soil_tempComponent()
{
       
}


double soil_tempComponent::getair_temp_day1(){ return this->air_temp_day1; }
std::vector<int> & soil_tempComponent::getlayer_thick(){ return this->layer_thick; }

void soil_tempComponent::setair_temp_day1(double _air_temp_day1)
{
    _Temp_profile.setair_temp_day1(_air_temp_day1);
}
void soil_tempComponent::setlayer_thick(const std::vector<int> & _layer_thick)
{
    _Temp_profile.setlayer_thick(_layer_thick);
    _Layers_temp.setlayer_thick(_layer_thick);
}
void soil_tempComponent::Calculate_Model(soil_tempState &s, soil_tempState &s1, soil_tempRate &r, soil_tempAuxiliary &a, soil_tempExogenous &ex)
{
    _Temp_amp.Calculate_Model(s, s1, r, a, ex);
    _Canopy_temp_avg.Calculate_Model(s, s1, r, a, ex);
    _Temp_profile.Calculate_Model(s, s1, r, a, ex);
    _Layers_temp.Calculate_Model(s, s1, r, a, ex);
    _Update.Calculate_Model(s, s1, r, a, ex);
}
soil_tempComponent::soil_tempComponent(soil_tempComponent& toCopy)
{
    air_temp_day1 = toCopy.getair_temp_day1();
    for (int i = 0; i < toCopy.getlayer_thick().size(); i++)
{
    layer_thick[i] = toCopy.getlayer_thick()[i];
}

}